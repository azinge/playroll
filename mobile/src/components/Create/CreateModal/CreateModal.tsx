import React from "react";
import { Text } from "native-base";
import RNPickerSelect from 'react-native-picker-select';
import {
  View,
  Alert,
  Modal,
  TouchableHighlight,
  SegmentedControlIOS,
  Image,
  TextInput
} from "react-native";
import styles, { pickerStyle } from "./CreateModal.styles";

export default class CreateModal extends React.Component<any, any> {
  render() {
    return (
      <Modal
          animationType="fade"
          transparent={true}
          visible={this.props.modalVisible}
          onRequestClose={() => {
            Alert.alert("Modal has been closed.");
          }}
        >
        <View style={styles.modalposition}>
          <View style={styles.modalbackground}>
            <Text style={styles.formtitle}>Add to Playroll</Text>
            <View style={styles.modaldata}>
              <Image 
                style={{width: 200, height: 200, borderRadius: 5 }}
                source={{uri: 'https://thefader-res.cloudinary.com/private_images/w_1260,c_limit,f_auto,q_auto:best/C9H8-PWUIAAzbQ2_j7a67b/full-credits-kendrick-lamar-damn.jpg'}}
              />
            </View>
            <Text style={styles.welcome}>Hip-Hop</Text>
            <View style={{width: 200, height: 100, marginLeft: 'auto', marginRight: 'auto', marginBottom: 0}}>
              <SegmentedControlIOS
                tintColor={"#6A0070"}
                values={["Popular", "Random"]}
                // selectedIndex={this.state.playFrom}
                onChange={event => {
                  this.setState({
                    selectedIndex: event.nativeEvent.selectedSegmentIndex
                  });
                }}
              />
              <View style={{display: 'flex', flexDirection: 'row',}}>
                <TextInput
                    // ref={(el) => {
                    //     this.inputRefs.name = el;
                    // }}
                    placeholder={'Mai = Waifu'}
                    returnKeyType="next"
                    enablesReturnKeyAutomatically
                    // onSubmitEditing={() => {
                        // this.inputRefs.picker.togglePicker();
                    // }}
                    style={pickerStyle.inputIOS}
                    blurOnSubmit={false}
                />
                <RNPickerSelect
                  placeholder={{ label: 'Minutes...', value: null,}}
                  items={[{label: 'Songs', value: 'songs'}, {label: 'Minutes', value: 'minutes'}]}
                  onValueChange={(value) => {
                      this.setState({
                          favColor: value,
                      });
                  }}
                  
                  // onUpArrow={() => {
                  //     this.inputRefs.name.focus();
                  // }}
                  // onDownArrow={() => {
                  //     this.inputRefs.picker2.togglePicker();
                  // }}
                  style={pickerStyle} //weird
                  // value={this.state.favColor}
                  // ref={(el) => {
                  //     this.inputRefs.picker = el;
                  // }}
              />
              </View>  
            </View>
            <View style={styles.formfooter}>
              <TouchableHighlight
                style={{marginLeft: 20}}
                onPress={() => {
                  // this.props.setModalVisible(!this.props.modalVisible);
                }}
              >
                <Text>OK</Text>
              </TouchableHighlight>

              <TouchableHighlight
                style={{marginRight: 20}}
                onPress={() => {
                  this.props.setModalVisible(!this.props.modalVisible);
                }}
              >
                <Text>Close</Text>
              </TouchableHighlight>
              
            </View>
           
          </View>
        </View>
      </Modal>
    )
  }
}