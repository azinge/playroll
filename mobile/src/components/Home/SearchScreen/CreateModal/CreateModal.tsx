import React from "react";
import RNPickerSelect from "react-native-picker-select";
import {
  View,
  Text,
  Alert,
  Modal,
  TouchableHighlight,
  SegmentedControlIOS,
  Image,
  TextInput,
} from "react-native";
import styles, { pickerStyle } from "./CreateModal.styles";

import { MusicSource } from "../../../../graphql/types";

import { CreateRollMutation } from "../../../../graphql/requests/Roll";
import { GET_PLAYROLL } from "../../../../graphql/requests/Playroll/GetPlayrollQuery";

export interface Props {
  currentSource?: MusicSource;
  modalVisible?: boolean;
  closeModal?: (redirect?: boolean) => void;
  manageRoll?: () => void;
  playrollID?: number;
}

export default class CreateModal extends React.Component<Props> {
  render() {
    const {
      currentSource = {},
      modalVisible = true,
      playrollID = 0,
      closeModal = () => {},
      manageRoll = () => {},
    } = this.props;
    return (
      <Modal
        animationType="fade"
        transparent={true}
        visible={modalVisible}
        onRequestClose={() => {
          Alert.alert("Modal has been closed.");
        }}
      >
        <View style={styles.modalposition}>
          <View style={styles.modalbackground}>
            <Text style={styles.formtitle}>Add to Playroll</Text>
            <View style={styles.modaldata}>
              <Image
                style={{ width: 200, height: 200, borderRadius: 5 }}
                source={{
                  uri: currentSource.cover,
                }}
              />
            </View>
            <Text style={styles.welcome}>{currentSource.name}</Text>
            <Text style={styles.welcome}>{currentSource.type}</Text>

            <View
              style={{
                width: 200,
                height: 100,
                marginLeft: "auto",
                marginRight: "auto",
                marginBottom: 0,
              }}
            >
              <SegmentedControlIOS
                tintColor={"#6A0070"}
                values={["Popular", "Random"]}
                // selectedIndex={this.state.playFrom}
                onChange={event => {
                  this.setState({
                    selectedIndex: event.nativeEvent.selectedSegmentIndex,
                  });
                }}
              />
              <View style={{ display: "flex", flexDirection: "row" }}>
                <TextInput
                  // ref={(el) => {
                  //     this.inputRefs.name = el;
                  // }}
                  placeholder={"Mai = Waifu"}
                  returnKeyType="next"
                  enablesReturnKeyAutomatically
                  // onSubmitEditing={() => {
                  // this.inputRefs.picker.togglePicker();
                  // }}
                  style={pickerStyle.inputIOS}
                  blurOnSubmit={false}
                />
                <RNPickerSelect
                  placeholder={{ label: "Minutes...", value: null }}
                  items={[
                    { label: "Songs", value: "songs" },
                    { label: "Minutes", value: "minutes" },
                  ]}
                  onValueChange={value => {
                    this.setState({
                      favColor: value,
                    });
                  }}
                  style={pickerStyle}
                />
              </View>
            </View>
            <View style={styles.formfooter}>
              <CreateRollMutation
                variables={{
                  input: {
                    playrollID: playrollID,
                    data: { sources: [currentSource] },
                  },
                }}
                onCompleted={() => {
                  closeModal(true);
                }}
                refetchQueries={[GET_PLAYROLL]}
              >
                {(createRoll, { data }) => (
                  <TouchableHighlight
                    style={{ marginLeft: 20 }}
                    onPress={() => {
                      playrollID ? createRoll() : manageRoll();
                    }}
                  >
                    <Text>{playrollID ? "Add" : "Continue"}</Text>
                  </TouchableHighlight>
                )}
              </CreateRollMutation>

              <TouchableHighlight
                style={{ marginRight: 20 }}
                onPress={() => {
                  closeModal();
                }}
              >
                <Text>Close</Text>
              </TouchableHighlight>
            </View>
          </View>
        </View>
      </Modal>
    );
  }
}
