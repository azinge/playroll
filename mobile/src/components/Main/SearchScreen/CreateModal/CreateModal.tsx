import React from 'react';
import RNPickerSelect from 'react-native-picker-select';
import {
  Text,
  View,
  Alert,
  Modal,
  TouchableHighlight,
  Image,
} from 'react-native';
import styles, { pickerStyle } from './CreateModal.styles';

import { MusicSource, RollFilter } from '../../../../graphql/types';

import { CreateRollMutation } from '../../../../graphql/requests/Roll';
import { GET_CURRENT_USER_PLAYROLL } from '../../../../graphql/requests/Playroll/GetCurrentUserPlayrollQuery';

export interface Props {
  currentSource: MusicSource;
  modalVisible: boolean;
  closeModal: (redirect?: boolean) => void;
  //   manageRoll: (currentSource?: MusicSource) => void;
  playrollID: number;
}

interface State {
  filter?: RollFilter;
  length?: RollFilter;
}

export default class CreateModal extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = {
      filter: undefined,
      length: undefined,
    };
  }

  render() {
    return (
      <Modal
        animationType='fade'
        transparent={true}
        visible={this.props.modalVisible}
        onRequestClose={() => {
          Alert.alert('Modal has been closed.');
        }}
      >
        <View style={styles.modalposition}>
          <View style={styles.modalbackground}>
            <Text style={styles.formtitle}>Add to Playroll</Text>
            <View style={styles.modaldata}>
              <Image
                style={{ width: 200, height: 200, borderRadius: 5 }}
                source={{
                  uri: this.props.currentSource.cover,
                }}
              />
            </View>
            <Text style={styles.welcome}>{this.props.currentSource.name}</Text>
            <Text style={styles.welcome}>{this.props.currentSource.type}</Text>

            <View
              style={{
                width: 200,
                height: 120,
                marginLeft: 'auto',
                marginRight: 'auto',
                marginBottom: 0,
              }}
            >
              <RNPickerSelect
                placeholder={{ label: 'Select Filter...', value: undefined }}
                items={[
                  { label: 'Play Songs In Order', value: 'In Order' },
                  { label: 'Play Songs In Random Order', value: 'Random' },
                ]}
                onValueChange={value => {
                  switch (value) {
                    case 'In Order':
                      this.setState({
                        filter: { type: 'In Order' },
                      });
                      break;
                    case 'Random':
                      this.setState({
                        filter: { type: 'Random' },
                      });
                      break;
                    default:
                      this.setState({
                        filter: undefined,
                      });
                  }
                }}
                style={pickerStyle}
              />
              <RNPickerSelect
                placeholder={{ label: 'Select Length...', value: undefined }}
                items={[
                  { label: 'Play All Songs', value: 'Original' },
                  { label: 'Play 5 Songs', value: 'Number' },
                ]}
                onValueChange={value => {
                  switch (value) {
                    case 'Original':
                      this.setState({
                        length: { type: 'Original' },
                      });
                      break;
                    case 'Number':
                      this.setState({
                        length: { type: 'Number', modifications: ['0', '5'] },
                      });
                      break;
                    default:
                      this.setState({
                        length: undefined,
                      });
                  }
                }}
                style={pickerStyle}
              />
            </View>
            <View style={styles.formfooter}>
              <CreateRollMutation
                variables={{
                  input: {
                    playrollID: this.props.playrollID,
                    data: {
                      sources: [this.props.currentSource],
                      filters: [this.state.filter, this.state.length],
                    },
                  },
                }}
                onCompleted={() => {
                  this.props.closeModal(true);
                }}
                refetchQueries={() => [GET_CURRENT_USER_PLAYROLL]}
              >
                {(createRoll, { data }) => (
                  <TouchableHighlight
                    style={{ marginLeft: 20 }}
                    onPress={() => {
                      createRoll();
                    }}
                  >
                    <Text>{this.props.playrollID ? 'Add' : 'Continue'}</Text>
                  </TouchableHighlight>
                )}
              </CreateRollMutation>

              <TouchableHighlight
                style={{ marginRight: 20 }}
                onPress={() => {
                  this.props.closeModal();
                }}
              >
                <Text>Close</Text>
              </TouchableHighlight>
            </View>
          </View>
        </View>
      </Modal>
    );
  }
}
