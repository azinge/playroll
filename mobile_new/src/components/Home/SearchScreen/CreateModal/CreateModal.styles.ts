import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  container: {
      flex: 1,
      justifyContent: 'center',
      alignItems: 'center',
      backgroundColor: '#F5FCFF',
  },
  welcome: {
      fontSize: 20,
      color: '#6A0070',
      marginLeft: 50,
      // textAlign: 'center',
      // margin: 100,
  },
  instructions: {
      textAlign: 'center',
      color: '#333333',
      marginBottom: 5,

  },
  modaldata: {
    marginLeft: 50,
    marginRight: 50,
    marginTop: 10,
    marginBottom: 10,
    shadowColor: 'black',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.3,
  },
  footercontent: {
      backgroundColor: '#00ffff',
      flexDirection: 'row',
  },
  footerbutton: {
      justifyContent: 'center',
      alignItems: 'center',
      marginTop: 5,
      marginBottom: 2,
      width: '33%',
  },
  formtitle: {
    fontSize: 24,
    color: '#6A0070',
    marginTop: 20,
    marginLeft: 15,
  },
  formfooter: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'space-between',
    bottom: 10,
  },
  modalposition: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'rgba(52, 52, 52, 0.6)',
  },
  modalbackground: {
    flex: .6,
    flexDirection: 'column',
    borderRadius: 20,
    backgroundColor: 'white',
    position: 'absolute',
    // margin: 50
  },
});

export const pickerStyle = StyleSheet.create({
  inputIOS: {
    marginTop: 10,
    width: 100,
    height: 40,
    // margin: 10,
    fontSize: 12,
    paddingTop: 13,
    paddingHorizontal: 10,
    paddingBottom: 12,
    borderWidth: 1,
    borderColor: '#6A0070',
    borderRadius: 4,
    backgroundColor: 'white',
    color: 'black',
    alignItems: 'flex-end',
  },
});
