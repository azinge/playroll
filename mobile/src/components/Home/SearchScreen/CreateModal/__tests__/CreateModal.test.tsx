import React from "react";
import renderer from "react-test-renderer";
import { MockedProvider } from "react-apollo/test-utils";

import CreateModal from "../CreateModal";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <MockedProvider>
        <CreateModal />
      </MockedProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
