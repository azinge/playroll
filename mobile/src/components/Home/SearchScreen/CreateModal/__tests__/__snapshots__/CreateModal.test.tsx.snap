// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={true}
  visible={false}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(52, 52, 52, 0.6)",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 20,
          "flex": 0.6,
          "flexDirection": "column",
          "position": "absolute",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#6A0070",
            "fontSize": 24,
            "marginLeft": 15,
            "marginTop": 20,
          }
        }
      >
        Add to Playroll
      </Text>
      <View
        style={
          Object {
            "marginBottom": 10,
            "marginLeft": 50,
            "marginRight": 50,
            "marginTop": 10,
            "shadowColor": "black",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.3,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": undefined,
            }
          }
          style={
            Object {
              "borderRadius": 5,
              "height": 200,
              "width": 200,
            }
          }
        />
      </View>
      <Text
        style={
          Object {
            "color": "#6A0070",
            "fontSize": 20,
            "marginLeft": 50,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "#6A0070",
            "fontSize": 20,
            "marginLeft": 50,
          }
        }
      />
      <View
        style={
          Object {
            "height": 100,
            "marginBottom": 0,
            "marginLeft": "auto",
            "marginRight": "auto",
            "width": 200,
          }
        }
      >
        <RCTSegmentedControl
          enabled={true}
          onChange={[Function]}
          style={
            Array [
              Object {
                "height": 28,
              },
              undefined,
            ]
          }
          tintColor="#6A0070"
          values={
            Array [
              "Popular",
              "Random",
            ]
          }
        />
        <View
          style={
            Object {
              "display": "flex",
              "flexDirection": "row",
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            blurOnSubmit={false}
            enablesReturnKeyAutomatically={true}
            placeholder="Mai = Waifu"
            returnKeyType="next"
            style={
              Object {
                "alignItems": "flex-end",
                "backgroundColor": "white",
                "borderColor": "#6A0070",
                "borderRadius": 4,
                "borderWidth": 1,
                "color": "black",
                "fontSize": 12,
                "height": 40,
                "marginTop": 10,
                "paddingBottom": 12,
                "paddingHorizontal": 10,
                "paddingTop": 13,
                "width": 100,
              }
            }
            underlineColorAndroid="transparent"
          />
          <View
            style={
              Array [
                Object {
                  "alignSelf": "stretch",
                },
                undefined,
              ]
            }
          >
            <View
              accessible={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              pointerEvents="box-only"
            >
              <TextInput
                allowFontScaling={true}
                editable={false}
                style={
                  Array [
                    Object {
                      "paddingRight": 30,
                    },
                    Object {
                      "alignItems": "flex-end",
                      "backgroundColor": "white",
                      "borderColor": "#6A0070",
                      "borderRadius": 4,
                      "borderWidth": 1,
                      "color": "black",
                      "fontSize": 12,
                      "height": 40,
                      "marginTop": 10,
                      "paddingBottom": 12,
                      "paddingHorizontal": 10,
                      "paddingTop": 13,
                      "width": 100,
                    },
                    Object {
                      "color": "#C7C7CD",
                    },
                  ]
                }
                underlineColorAndroid="transparent"
                value="Minutes..."
              />
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderLeftColor": "transparent",
                      "borderLeftWidth": 10,
                      "borderRightColor": "transparent",
                      "borderRightWidth": 10,
                      "borderTopColor": "gray",
                      "borderTopWidth": 10,
                      "height": 0,
                      "position": "absolute",
                      "right": 10,
                      "top": 20,
                      "width": 0,
                    },
                    undefined,
                  ]
                }
                testID="icon_ios"
              />
            </View>
            <Modal
              hardwareAccelerated={false}
              supportedOrientations={
                Array [
                  "portrait",
                  "landscape",
                ]
              }
              testID="RNPickerSelectModal"
              transparent={true}
              visible={false}
            >
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "opacity": 1,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "#EFF1F2",
                      "borderTopColor": "#919498",
                      "borderTopWidth": 0.5,
                      "flexDirection": "row",
                      "height": 44,
                      "justifyContent": "space-between",
                      "zIndex": 2,
                    },
                    undefined,
                  ]
                }
                testID="done_bar"
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "row",
                      "marginLeft": 15,
                    }
                  }
                >
                  <View
                    accessible={true}
                    isTVSelectable={true}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "backgroundColor": "transparent",
                            "borderRightColor": "#D0D4DB",
                            "borderRightWidth": 1.5,
                            "borderTopColor": "#D0D4DB",
                            "borderTopWidth": 1.5,
                            "height": 15,
                            "width": 15,
                          },
                          undefined,
                          Object {
                            "transform": Array [
                              Object {
                                "translateY": 17,
                              },
                              Object {
                                "rotate": "-45deg",
                              },
                            ],
                          },
                          undefined,
                          Object {},
                        ]
                      }
                    />
                  </View>
                  <View
                    style={
                      Object {
                        "marginHorizontal": 10,
                      }
                    }
                  />
                  <View
                    accessible={true}
                    isTVSelectable={true}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "backgroundColor": "transparent",
                            "borderRightColor": "#D0D4DB",
                            "borderRightWidth": 1.5,
                            "borderTopColor": "#D0D4DB",
                            "borderTopWidth": 1.5,
                            "height": 15,
                            "width": 15,
                          },
                          undefined,
                          Object {
                            "transform": Array [
                              Object {
                                "translateY": 8,
                              },
                              Object {
                                "rotate": "135deg",
                              },
                            ],
                          },
                          undefined,
                          Object {},
                        ]
                      }
                    />
                  </View>
                </View>
                <View
                  accessible={true}
                  hitSlop={
                    Object {
                      "bottom": 2,
                      "left": 2,
                      "right": 2,
                      "top": 2,
                    }
                  }
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  testID="done_button"
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#007AFE",
                          "fontSize": 18,
                          "fontWeight": "bold",
                          "padding": 10,
                        },
                        undefined,
                      ]
                    }
                  >
                    Done
                  </Text>
                </View>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "#D0D4DB",
                      "height": 215,
                      "justifyContent": "center",
                    },
                    undefined,
                  ]
                }
              >
                <View>
                  <RCTPicker
                    items={
                      Array [
                        Object {
                          "label": "Minutes...",
                          "textColor": undefined,
                          "value": null,
                        },
                        Object {
                          "label": "Songs",
                          "textColor": undefined,
                          "value": "songs",
                        },
                        Object {
                          "label": "Minutes",
                          "textColor": undefined,
                          "value": "minutes",
                        },
                      ]
                    }
                    onChange={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    selectedIndex={0}
                    style={
                      Array [
                        Object {
                          "height": 216,
                        },
                        undefined,
                      ]
                    }
                  />
                </View>
              </View>
            </Modal>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "bottom": 10,
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "marginLeft": 20,
            }
          }
        >
          <Text
            style={null}
          >
            Continue
          </Text>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "marginRight": 20,
            }
          }
        >
          <Text
            style={null}
          >
            Close
          </Text>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
